packages/shared/lib/utils/utils.ts

```diff
@@ -117,13 +117,14 @@ export function getRedisUrl() {
 export function isValidHttpUrl(str: string) {
     const pattern = new RegExp(
         '^(https?:\\/\\/)?' + // protocol
-            '((([a-z\\d]([a-z\\d-]*[a-z\\d])*)\\.)+[a-z]{2,}|localhost|' + // domain name
+            '((([a-z\\d]([a-z\\d-]*[a-z\\d])*)\\.)+[a-z]{2,}|' + // domain name
+            'localhost|' + // OR localhost
             '((\\d{1,3}\\.){3}\\d{1,3}))' + // OR ip (v4) address
-            '(\\:\\d+)?(\\/[-a-z\\d%_.~+]*)*' + // port and path
-            '(\\?[;&a-z\\d%_.~+=-]*)?' + // query string
-            '(\\#[-a-z\\d_]*)?$',
+            '(\\:\\d+)?(\\/[-a-z\\d%_.~+@]*)*' + // port and path
+            '(\\?[;&a-z\\d%_.~+=-@,]*)?' + // query string, including commas
+            '(\\#[-a-z\\d_]*)?$', // fragment locator
         'i'
-    ); // fragment locator
+    );
     return !!pattern.test(str);
 }
```

Comment:
I think you could remove all of this in favor of native parsing, wdyt?

```js
try { 
  new URL(str);
  return true;
} catch {
  return false;
}
```

Reply:
ah well @TBonnin was faster :D

Reply:
haha, thanks, yeah, updated with https://github.com/NangoHQ/nango/pull/1638/commits/30851a46906bf3c376ffa38712453f42dc67e9ea